src/error.rs

```diff
@@ -206,9 +211,20 @@ impl Error {
         self.inner.cause
     }
 
+    /// Returns the info of the client connection on which this error occurred.
+    #[cfg(all(feature = "client", any(feature = "http1", feature = "http2")))]
+    pub fn client_connect_info(&self) -> Option<&Connected> {
```

Comment:
I realize they're not implemented yet, but this made me think of the errors RFCs about "generic context" access, and would seem like a good fit eventually. https://github.com/rust-lang/rfcs/pull/2895 and https://github.com/rust-lang/rfcs/pull/3192